File: src/test/java/com/mikea/bayes/belief/CliqueTreeQueryAlgorithmTest.java
package com.mikea.bayes.belief;

import com.mikea.bayes.BayesianNetwork;
import com.mikea.bayes.Factor;
import com.mikea.bayes.SumProduct;
import com.mikea.bayes.Var;
import com.mikea.bayes.VarAssignment;
import com.mikea.bayes.data.StudentsNetwork;
import com.mikea.bayes.query.QueryAlgorithm;
import com.mikea.bayes.query.VarElimination;
import org.junit.Test;

import static com.mikea.bayes.VarSet.newVarSet;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

/**
 * @author mike.aizatsky@gmail.com
 */
public class CliqueTreeQueryAlgorithmTest {
    @Test
    public void testNoEvidence() throws Exception {
        QueryAlgorithm<?> baselineAlgorithm = new VarElimination(new SumProduct.MinNeighborsStrategy());
        QueryAlgorithm<?> algorithm = new CliqueTreeQueryAlgorithm();

        BayesianNetwork network = StudentsNetwork.buildStudentsNetwork();

        QueryAlgorithm.Result baselineResult = baselineAlgorithm.prepare(network);
        QueryAlgorithm.Result actualResult = algorithm.prepare(network);
        double epsilon = 0.001;

        for (Var var : network.getVarList()) {
            Factor expected = baselineResult.query(newVarSet(var));
            Factor actual = actualResult.query(newVarSet(var));
            assertTrue(expected.equals(actual, epsilon));

            double baselineProbability = baselineResult.getProbability(new VarAssignment(new Var[]{var}, new int[]{0}));
            double actualProbability = actualResult.getProbability(new VarAssignment(new Var[]{var}, new int[]{0}));
            assertEquals(baselineProbability, actualProbability, epsilon);
        }
    }
}

File: src/test/java/com/mikea/bayes/belief/CliqueTreeQueryAlgorithmTest.scala
package com.mikea.bayes.belief

import org.junit.Test

/**
 * @author mike.aizatsky@gmail.com
 */
class CliqueTreeQueryAlgorithmTest {
  @Test def testNoEvidence {
    ???
/*
    val baselineAlgorithm: QueryAlgorithm[_] = new VarElimination(new SumProduct.MinNeighborsStrategy)
    val algorithm: QueryAlgorithm[_] = new CliqueTreeQueryAlgorithm
    val network: BayesianNetwork = StudentsNetwork.buildStudentsNetwork
    val baselineResult: Nothing = baselineAlgorithm.prepare(network)
    val actualResult: Nothing = algorithm.prepare(network)
    val epsilon: Double = 0.001
    import scala.collection.JavaConversions._
    for (`var` <- network.varList) {
      val expected: Factor = baselineResult.query(newVarSet(`var`))
      val actual: Factor = actualResult.query(newVarSet(`var`))
      assertTrue(expected.equals(actual, epsilon))
      val baselineProbability: Double = baselineResult.getProbability(new VarAssignment(Array[Var](`var`), Array[Int](0)))
      val actualProbability: Double = actualResult.getProbability(new VarAssignment(Array[Var](`var`), Array[Int](0)))
      assertEquals(baselineProbability, actualProbability, epsilon)
    }
*/
  }
}